xsl/xccdf-resources/openscap.js

```diff
@@ -108,12 +108,131 @@ function ruleSearch()
         $("#search-matches").html("No rules match your search criteria!");
 }
 
+var is_original = true;
+var original_treetable = null;
+
 $(document).ready( function() {
     $("#result-details").hide();
     $(".js-only").show();
     $(".toggle-rule-display").each(function(){
         toggleRuleDisplay(this);
     });
-
-    $(".treetable").treetable({ column: 0, expandable: true, initialState : "expanded", indent : 0 });
+    original_treetable = $(".treetable").clone();
+    $(".treetable").treetable({ column: 0, expandable: true, clickableNodeNames: true, initialState : "expanded", indent : 0 });
+    is_original = true;
 });
+
+function Reset() {
+	if (!is_original) {
+		$(".treetable").remove();
+		$("#rule-overview").append(original_treetable.clone());
+		$(".treetable").treetable({ column: 0, expandable: true, clickableNodeNames: true, initialState : "expanded", indent : 0 });
+		$(".toggle-rule-display").each(function(){
+			toggleRuleDisplay(this);
+		});
+		is_original = true;
+	}
+}
+
+function NewGroupLine(group_name)
+{
+	return "<tr class=\"rule-overview-inner-node\" data-tt-id=\"" + group_name + "\">" +
+		"<td colspan=\"3\"><strong>" + group_name + "</strong></td></tr>";
```

Comment:
jQuery has some functions which can create new elements. Maybe it would be cleaner way.